src/PROPOSAL/PROPOSAL/crosssection/CrossSectionDirect.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "PROPOSAL/crosssection/CrossSection.h"
+#include "PROPOSAL/crosssection/parametrization/ParametrizationDirect.h"
+
+namespace PROPOSAL {
+    class CrossSectionDirect : public CrossSectionBase {
+    public:
+        template <typename Param,
+                typename _name = crosssection::ParametrizationName<Param>>
+        CrossSectionDirect(Param param_, ParticleDef p,
```

Comment:
The design concept, here, is something, I think, can be improved. But I would like a broader discussion about that.

PROPOSAL initially only had interactions differential in the energy loss, so they always had to be integrated or interpolated for a dEdx or dNdx.
This changed, when the weak interaction or annihilation were introduced, which are only stochastic and don't have continuous losses. However, both interactions still have to be integrated or ininterpolated, since they are given differential  on how the energy is separated between the secondary particles.
To make them adaptable, they got an dEdx=0 or dE2dx=0 and still some energy cuts, which were not used.

Now with this new sort of interaction, the photoproduction, also the interpolation gets obsolet, raising the question if it's time for some restructuring.
There should be a class of only stochastic cross sections with minimal overhead, necessary for this photoproduction. Derived from that, a cross section class still consisting only of stochastic interaction, but which are differential in partitioning the energy to the secondaries and therefor neds to be integrated for the total cross section. (Btw. also the CSMS parametrization of the weak interaction have the total cross sections also tabulated, so they don't need to be integrated, which might be more stable.)

In this approach the interactions, which are able to create continuous energy losses, and the interactions, where the primary particle dosen't exist anymore after the interaction (where also the decay might be considered) gets clearly separated.

ps: this might also help in the long term TODO to perform only-stochastic propagations, e.g. for neutrinos.